pydocumentdb/auth.py

```diff
@@ -67,11 +69,25 @@ def __GetAuthorizationTokenUsingMasterKey(verb,
         resource_id_or_fullname=(resource_id_or_fullname or ''),
         x_date=headers.get(http_constants.HttpHeaders.XDate, '').lower(),
         http_date=headers.get(http_constants.HttpHeaders.HttpDate, '').lower())
-   
-    body = text.decode('utf8')
+
+    try:
```

Comment:
@RedCraig Would you mind, separating the Python 3 changes for this PR into a separate one, so that more people can review it independently of the test changes? I'll have it reviewed by couple other folks. This applies to the changes in all other files as part of this PR.